src/routes/(app)/search/+page.server.ts

```diff
@@ -1,14 +1,45 @@
+import type { Hits } from "meilisearch";
 import type { Actions } from "./$types";
+import type { SearchDataWithType } from "$lib/search/searchTypes";
 
 export const actions = {
   default: async (event) => {
     const data = await event.request.formData();
-    const search = data.get("input");
-    if (typeof search !== "string") return;
-    const response = await event.fetch(
-      "/api/members?" + new URLSearchParams({ search }),
-    );
-    const users = await response.json();
-    return { users };
+
+    const query = data.get("input");
+    if (typeof query !== "string") return;
+
+    const includeMembers = data.get("members") === "on";
+    const includeEvents = data.get("events") === "on";
+    const includeArticles = data.get("articles") === "on";
+    const includePositions = data.get("positions") === "on";
+    const includeSongs = data.get("songs") === "on";
+    const indexes = [];
+    if (includeMembers) indexes.push("members");
+    if (includeEvents) indexes.push("events");
+    if (includeArticles) indexes.push("articles");
+    if (includePositions) indexes.push("positions");
+    if (includeSongs) indexes.push("songs");
+
+    const language = data.get("language") === "en" ? "en" : "sv";
+
+    const url = new URL("/api/search", event.request.url);
+    url.searchParams.set("query", query);
+    url.searchParams.set("indexes", JSON.stringify(indexes));
+    url.searchParams.set("language", language);
+    const response = await event.fetch(url);
+    if (!response.ok) {
+      // silently fail
+      return;
+    }
```

Comment:
Who no error handling? We should at least log errors so we can investigate any problems in the future

Reply:
Could we either just log the error server side, or add error handling client side? Otherwise we just get the generic 500 internal server error which is not very nice

Reply:
Fixed in latest push!